pkg/cmd/cli/cmd/create_route.go

```diff
@@ -0,0 +1,396 @@
+	cmd.Flags().String("port", "", "Name of the service port or number of the container port the route will route traffic to")
+	cmd.Flags().String("expose", "", "Name of the service that the new route is exposing")
+	cmd.MarkFlagRequired("expose")
+	cmd.Flags().String("path", "", "Path that the router watches to route traffic to the service.")
+	cmd.Flags().String("cert", "", "Path to a certificate file.")
+	cmd.Flags().String("key", "", "Path to a key file.")
+	cmd.Flags().String("ca-cert", "", "Path to a CA certificate file.")
+	cmd.Flags().String("dest-ca-cert", "", "Path to a CA certificate file, used for securing the connection from the router to the destination.")
+	cmd.MarkFlagRequired("dest-ca-cert")
+
+	return cmd
+}
+
+// CreateReencryptRoute implements the behavior to run the create reencrypt route command.
+func CreateReencryptRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	route, err := unsecuredRoute(cmd, kc, ns, args)
+	if err != nil {
+		return err
+	}
+
+	route.Spec.Host = kcmdutil.GetFlagString(cmd, "hostname")
+	route.Spec.Path = kcmdutil.GetFlagString(cmd, "path")
+
+	route.Spec.TLS = new(api.TLSConfig)
+	route.Spec.TLS.Termination = api.TLSTerminationEdge
+
+	cert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "cert-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Certificate = string(cert)
+	key, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "key-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.Key = string(key)
+	caCert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "ca-cert-file"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.CACertificate = string(caCert)
+	destCACert, err := fileutil.LoadData(kcmdutil.GetFlagString(cmd, "dest-ca-cert"))
+	if err != nil {
+		return err
+	}
+	route.Spec.TLS.DestinationCACertificate = string(destCACert)
+
+	route, err = oc.Routes(ns).Create(route)
+	if err != nil {
+		return err
+	}
+	mapper, typer := f.Object()
+	resourceMapper := &resource.Mapper{
+		ObjectTyper:  typer,
+		RESTMapper:   mapper,
+		ClientMapper: resource.ClientMapperFunc(f.Factory.RESTClient),
+	}
+	info, err := resourceMapper.InfoForObject(route)
+	if err != nil {
+		return err
+	}
+	shortOutput := kcmdutil.GetFlagString(cmd, "output") == "name"
+	kcmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "created")
+	return nil
+}
+
+// unsecuredRoute will return a route with enough info so that it can direct traffic to
+// the service provided by --expose. Callers of this helper are responsible for providing
+// tls configuration, path, and the hostname of the route.
+func unsecuredRoute(cmd *cobra.Command, kc *kclient.Client, namespace string, args []string) (*api.Route, error) {
+	serviceName := kcmdutil.GetFlagString(cmd, "expose")
```

Comment:
> Is this a hard requirement? I need to bring any kind of machinery in unsecuredRoute for this and it's ugly.

Yeah, we can argue about timing (if its really that hard, but I thought you wrote a helper already), but we output resources by "resource/name" all over our CLI.  This really needs to accept them to reduce friction in usage.